pytools/__init__.py

```diff
@@ -675,8 +675,9 @@ def memoize_on_first_arg(function, cache_dict_name=None):
     """
 
     if cache_dict_name is None:
-        cache_dict_name = intern("_memoize_dic_"
-                + function.__module__ + function.__name__)
+        cache_dict_name = intern(
+                "_memoize_dic_{function.__module__}{function.__name__}"
```

Comment:
Should this be an f-string?

Reply:
It really should! Thanks!

Should be fixed in latest version.